lua/nvim-treesitter/locals.lua

```diff
@@ -158,6 +140,42 @@ function M.recurse_local_nodes(local_def, accumulator, full_match, last_match)
   end
 end
 
+--- Get a single dimension table to look definition nodes by scope and text.
+-- This is memoized by buffer tick to avoid computing on every lookup.
+M.get_defintions_lookup_table = ts_utils.memoize_by_buf_tick(function(bufnr)
```

Comment:
Little typo here

Reply:
Furthermore, I don't see the point of exposing that if we don't expose the `get_definition_id` too.

Reply:
Last comment I promise, can you add a bit more doc regarding what represents the return value of this function, and maybe the idea behind it ? I don't find that especially easy to get.

Reply:
Yeah good point. I will update this.

Reply:
and it has a typo